src/static/js/chapter.js

```diff
@@ -256,6 +256,94 @@ function setDiscussionCount() {
   }
 }
 
+function indexHighlighter() {
+    //Only activate this if IntersectionObserver is supported
+  if(!('IntersectionObserver' in window)) {
+    gtag('event', 'index-highlighter', { 'event_category': 'user', 'event_label': 'not-enabled', 'value': 0 });
+    return;
+  }
+
+  var chapterIndex = document.querySelector('.index-box');
+
+  // Check if user has set reduced motion and only continue if not
+  var hasOSReducedMotion = window.matchMedia('(prefers-reduced-motion: reduce)').matches;
+  if (hasOSReducedMotion) {
+    console.log('User has set prefers-reduced-motion to ' + hasOSReducedMotion + ' so not highlighting the current section in chapter index');
+    gtag('event', 'prefers-reduced-motion', { 'event_category': 'user', 'event_label': 'reduce', 'value': 0 });
+    gtag('event', 'index-highlighter', { 'event_category': 'user', 'event_label': 'not-enabled', 'value': 0 });
+    return;
+  }
+
+  // Check if 'position:sticky' is supported (as this is not great UX when not so don't bother)
+  // Add the sticky class (which sets 'position:sticky') and then test if that stuck :-)
+  // Also use endsWith to support vendor prefixes (Safari v12 needs this)
+  chapterIndex && chapterIndex.classList.add('sticky');
+  var chapterIndexStyles = getComputedStyle(chapterIndex);
+  if (!chapterIndexStyles || !chapterIndexStyles.position || !chapterIndexStyles.position.endsWith('sticky')) {
+    gtag('event', 'index-highlighter', { 'event_category': 'user', 'event_label': 'not-enabled', 'value': 0 });
+    return
+  }
+
+  // Restrict the page height of the index to the page-height, as we're going to scroll this.
+  chapterIndex.classList.add('page-height');
+
+  // Create a function to handle highlighting a new index item
+  // that will be called by the IntersectionObserver
+  function highlightIndexEntry(link) {
+    
+    var indexLink = document.querySelector('.index-box a[href="#' + link + '"]');
+    var oldIndexLink = document.querySelector('.index-box .active');
+
+    if (indexLink && indexLink != oldIndexLink) {

```

Comment:
Invert and return to avoid nesting

Also using `isEqualNode`. Not sure if totally necessary but better to use builtins to be safe.

```suggestion
    if (!indexLink || indexLink. isEqualNode(oldIndexLink)) {
      return;
    }
```